api/v1/GameSession.php

```diff
@@ -78,10 +78,13 @@ public static function getRequestApiRootAsync(bool $forDocker = false): PromiseI
         $apiRoot = preg_replace('/(.*)\/(api|_profiler)\/(.*)/', '$1/', $_SERVER["REQUEST_URI"]);
         $apiRoot = str_replace("//", "/", $apiRoot);
 
-        // this is always called from inside the docker environment,so just use http://caddy:80/...
+
         if ($forDocker) {
             $deferred = new Deferred();
-            $GLOBALS['RequestApiRoot'][1] = 'http://caddy:80'.$apiRoot;
+            // this is always called from inside the docker environment,so just use http://caddy:80/ or
+            //   http://mitmproxy:8080/
+            $GLOBALS['RequestApiRoot'][1] = 'http://'.
+                ($_ENV['WEB_SERVER_HOST'] ?? 'caddy').':'.($_ENV['WEB_SERVER_PORT'] ?? 80).$apiRoot;
```

Comment:
please sift this out for a separate merge into dev so it can become part of 4.0.2